shared/util.c

```diff
@@ -237,32 +237,6 @@ ssize_t read_str_safe(int fd, char *buf, size_t buflen)
 	return done;
 }
 
-ssize_t write_str_safe(int fd, const char *buf, size_t buflen)
-{
-	size_t todo = buflen;
-	size_t done = 0;
-
-	assert_cc(EAGAIN == EWOULDBLOCK);
-
-	do {
-		ssize_t r = write(fd, buf + done, todo);
```

Comment:
Why did we remove the helper instead of adapting it to use `fwrite()`? AFAICT all the safety handling it does is still applicable.

Reply:
The problem with FILE output is that `fclose` itself could still write data. And this is a one-shot approach. We cannot call it multiple times until the remaining data is successfully written.

I mistakenly believed that FILE itself handles short writes, but I was wrong. So if we want to keep interrupt-safe writes, we cannot use FILE here. At best, we would adjust the API, but ... Not in a simple PR. :)